tests/unit/components/playback/controls/StimulationStudioWidget.spec.js

```diff
@@ -0,0 +1,161 @@
+    expect(well.at(0).attributes("fill")).toStrictEqual("#19AC8A");
+  });
+  test("Given the StimulationStudioWidget is in the range of 0 - 23, When mounted successfully, Then verify that Stimulation Plate Well is teal color and Alphabets A-X is visible", async () => {
+    const protocol_list = Array.from(Array(24).keys());
+
+    const propsData = {
+      protocol_code: protocol_list,
+    };
+    wrapper = mount(StimulationStudioWidget, {
+      propsData,
+      store,
+      localVue,
+    });
+    const well = wrapper.findAll("circle");
+    const protocol_name = wrapper.findAll(
+      ".span__simulationstudio-plate-well-protocol-location"
+    );
+    expect(well.at(0).attributes("fill")).toStrictEqual("#19AC8A");
+    expect(protocol_name.at(0).text()).toStrictEqual("A");
+    expect(well.at(23).attributes("fill")).toStrictEqual("#19AC8A");
+    expect(protocol_name.at(23).text()).toStrictEqual("X");
+  });
+  test("Given the StimulationStudioWidget is in the range of 24 - 48, When mounted successfully, Then verify that Stimulation Plate Well is teal color(at 0) and blue color(at 23) and Alphabets Y-V is visible", async () => {
+    const protocol_list = [];
+
+    for (let i = 24; i < 48; i++) protocol_list.push(i);
+    const propsData = {
+      protocol_code: protocol_list,
+    };
+    wrapper = mount(StimulationStudioWidget, {
+      propsData,
+      store,
+      localVue,
+    });
+    const well = wrapper.findAll("circle");
+    const protocol_name = wrapper.findAll(
+      ".span__simulationstudio-plate-well-protocol-location"
+    );
+    expect(well.at(0).attributes("fill")).toStrictEqual("#19AC8A");
+    expect(protocol_name.at(0).text()).toStrictEqual("Y");
+    expect(well.at(23).attributes("fill")).toStrictEqual("#005470");
+    expect(protocol_name.at(23).text()).toStrictEqual("V");
+  });
+  test("Given the StimulationStudioWidget is in the range of 48 - 71, When mounted successfully, Then verify that Stimulation Plate Well is blue color(at 0) and yellow color(at 23) and Alphabets W-V is visible", async () => {
+    const protocol_list = [];
+
+    for (let i = 48; i < 72; i++) protocol_list.push(i);
+    const propsData = {
+      protocol_code: protocol_list,
+    };
+    wrapper = mount(StimulationStudioWidget, {
+      propsData,
+      store,
+      localVue,
+    });
+    const well = wrapper.findAll("circle");
+    const protocol_name = wrapper.findAll(
+      ".span__simulationstudio-plate-well-protocol-location"
+    );
+    expect(well.at(0).attributes("fill")).toStrictEqual("#005470");
+    expect(protocol_name.at(0).text()).toStrictEqual("W");
+    expect(well.at(23).attributes("fill")).toStrictEqual("#f9d78c");
+    expect(protocol_name.at(23).text()).toStrictEqual("T");
+  });
+  test("Given the StimulationStudioWidget is in the range of 72 - 96, When mounted successfully, Then verify that Stimulation Plate Well is yellow color(at 0) and orange color(at 23) and Alphabets W-V is visible", async () => {
```

Comment:
similar comment above about testing the border/edge cases of your code logic

Reply:
Not essentially required VRT covers a good amount of all scenarios.